cli/ballerina-cli/src/main/java/io/ballerina/cli/cmd/ToolCommand.java

```diff
@@ -228,28 +240,76 @@ private void handlePullCommand() {
             }
         }
 
-        Path balaCacheDirPath = ProjectUtils.createAndGetHomeReposPath()
-                .resolve(REPOSITORIES_DIR).resolve(CENTRAL_REPOSITORY_CACHE_NAME)
-                .resolve(ProjectConstants.BALA_DIR_NAME);
+        pullToolAndUpdateBalToolsToml(toolId, version);
```

Comment:
Shall we separate out this into two methods and invoke them one after another? 
- pullTool
- updateBalToolsToml

Reply:
Within `pullToolAndUpdateBalToolsToml` there is `pullToolFromCentral` and `insertToBalToolsTomlFile` which does these tasks one after another. In addition, it contains error handling logic. However decided to have  `pullToolAndUpdateBalToolsToml` as a separate method because this piece of logic is used in 2 different places.

Reply:
Ack. Let's keep it as is then.